arkouda/pdarraycreation.py

```diff
@@ -285,24 +301,30 @@ def array(
         except TypeError:
             raise RuntimeError(f"Unhandled dtype {a.dtype}")
     else:
+        shape, ndim, full_size = _infer_from_size(a.shape)
+
         # Do not allow arrays that are too large
-        size = a.size
-        if (size * a.itemsize) > maxTransferBytes:
+        if (full_size * a.itemsize) > maxTransferBytes:
             raise RuntimeError(
                 "Array exceeds allowed transfer size. Increase ak.client.maxTransferBytes to allow"
             )
         # Pack binary array data into a bytes object with a command header
         # including the dtype and size. If the server has a different byteorder
         # than our numpy array we need to swap to match since the server expects
         # native endian bytes
+
         aview = _array_memview(a)
-        rep_msg = generic_msg(
-            cmd=f"array<{a.dtype.name},1>",
-            args={"shape": size},
-            payload=aview,
-            send_binary=True,
-        )
-        return create_pdarray(rep_msg) if dtype is None else akcast(create_pdarray(rep_msg), dtype)
+        if ndim == 1:
+            rep_msg = generic_msg(
+                cmd=f"array<{a.dtype.name},{ndim}>",
+                args={"shape": full_size},
+                payload=aview,
+                send_binary=True,
+            )
+            return create_pdarray(rep_msg) if dtype is None else akcast(create_pdarray(rep_msg), dtype)
+
+        else:
+            return xp.asarray(a, dtype=dtype)  # type: ignore
```

Comment:
Why not return a multidimensional pdarray here? 

I don't think the array-api's `Array` type needs to be a part of the regular arkouda API. As far as I understand, the vision was for `Array` to be a thin wrapper around a `pdarray` that implements the Array API specification, while `pdarray` itself could implement something more like numpy's API).

Reply:
I made some changes, let me know what you think.